### PR TITLE
GAWB-1031: Return workflow status info in listSubmissions endpoint

### PR DESCRIPTION
**Pull request by MatthewBemis**
firecloud-ui: https://github.com/broadinstitute/firecloud-ui/pull/522

Explain for the new query introduced:

`explain select s.id, w.status, count(*)
from SUBMISSION s join WORKFLOW w on s.id = w.submission_id
where s.workspace_id = 0x260d106c9227458d8936c5379c94da8b
group by s.id, w.status;`

screen shot 2016-09-27 at 2 21 49 pm
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  - Acceptance criteria exists and is met
  - Note any changes to implementation from the description
  - Add notes on what you've tested
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Database checks:
  - If PR includes new or changed db queries, include the explain plans in the description
  - Make sure liquibase is updated if appropriate
  - If doing a migration, take a backup of the
    [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
    and
    [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
    DBs in Google Cloud Console
- [x] **Submitter**: Update FISMA documentation if changes to:
  - Authentication
  - Authorization
  - Encryption
  - Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] Tell  that the PR exists if he wants to look at it **(apply requires_doge label)**
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
- Review cycle:
  - LR reviews
  - Rest of team may comment on PR at will
  - **LR assigns to submitter** for feedback fixes
  - Submitter rebases to develop again if necessary
  - Submitter makes further commits. DO NOT SQUASH
  - Submitter updates documentation as needed
  - Submitter **reassigns to LR** for further feedback
- [ ]  sign off
- [x] **LR** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/SubmissionComponent.scala [hunk 1]
```diff
@@ -311,6 +315,14 @@ trait SubmissionComponent {
       query.result.map(_.toMap)
     }
 
+    def getSubmissionWorkflowStatusCounts(submissionId: UUID): ReadAction[Map[String, Int]] = {
+      val query = for {
+        workflow <- workflowQuery if workflow.submissionId === submissionId
+      } yield (workflow.status)
+
+      query.result.map(wfs => wfs.groupBy(identity).mapValues(_.size))
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/SubmissionComponent.scala:324]: this pulls all the workflows into memory then groups and counts, that should all be done in the database

**Review by dvoet — CHANGES_REQUESTED** [review]
The code as written does a query per submission. This can be done instead as 1 query for all submissions. It is possible that this could be added to the starting query but that may be too complicated. If so, it can be a second query.

essentially the second query would be like
select s.id, w.status, count(*)
from submission s join workflow w on s.id = w.submission_id
where s.workspace_id = ?
group by s.id, w.status

**Comment by MatthewBemis** [comment]
@dvoet cc97fce reduces number of queries from n<sub>submission</sub> to 1

File: src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/SubmissionComponent.scala [hunk 2]
```diff
@@ -117,13 +117,18 @@ trait SubmissionComponent {
...
+        val states = workflowStates.groupBy(_.submissionId)
+        query.result.map { recs => recs.map {
           case (submissionRec, userRec, methodConfigRec, entityRec) =>
             val user = rawlsUserQuery.unmarshalRawlsUser(userRec)
             val config = methodConfigurationQuery.unmarshalMethodConfig(methodConfigRec, Map.empty, Map.empty, Map.empty)
             val entity = AttributeEntityReference(entityRec.entityType, entityRec.name)
-            val sub = unmarshalSubmission(submissionRec, config, entity, Seq.empty)
-            new SubmissionListResponse(sub, user)
+
+            val subStatuses = states.getOrElse(submissionRec.id, Seq.empty).map(x => x.workflowStatus -> x.count).toMap
```
Comment (MatthewBemis) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/SubmissionComponent.scala:128]: just realized I left a temp variable name "x" in here, i'll fix that tomorrow

**Review by dvoet — APPROVED** [review]
👍

**Comment by ansingh7115** [comment]
👍

**Pull request merged**